Source/WebKit/GPUProcess/media/RemoteAudioVideoRendererProxyManager.cpp

```diff
@@ -0,0 +1,576 @@
+{
+    if (RefPtr renderer = rendererFor(identifier))
+        renderer->contentBoxRectChanged(rect);
+}
+
+void RemoteAudioVideoRendererProxyManager::notifyWhenHasAvailableVideoFrame(WebKit::RemoteAudioVideoRendererIdentifier identifier, bool notify)
+{
+    if (RefPtr renderer = rendererFor(identifier)) {
+        if (notify) {
+            renderer->notifyWhenHasAvailableVideoFrame([weakThis = WeakPtr { *this }, identifier](auto presentationTime, auto clockTime) {
+                if (RefPtr protectedThis = weakThis.get()) {
+                    if (!protectedThis->m_renderers.contains(identifier))
+                        return;
+                    protectedThis->m_gpuConnectionToWebProcess.get()->connection().send(Messages::AudioVideoRendererRemoteMessageReceiver::HasAvailableVideoFrame(presentationTime, clockTime, protectedThis->stateFor(identifier)), identifier);
+                }
+            });
+            return;
+        }
+        renderer->notifyWhenHasAvailableVideoFrame({ });
+    }
+}
+
+void RemoteAudioVideoRendererProxyManager::expectMinimumUpcomingPresentationTime(RemoteAudioVideoRendererIdentifier identifier, const MediaTime& time)
+{
+    if (RefPtr renderer = rendererFor(identifier))
+        renderer->expectMinimumUpcomingPresentationTime(time);
+}
+
+void RemoteAudioVideoRendererProxyManager::setShouldDisableHDR(RemoteAudioVideoRendererIdentifier identifier, bool disable)
+{
+    if (RefPtr renderer = rendererFor(identifier))
+        renderer->setShouldDisableHDR(disable);
+}
+
+void RemoteAudioVideoRendererProxyManager::setPlatformDynamicRangeLimit(RemoteAudioVideoRendererIdentifier identifier, const WebCore::PlatformDynamicRangeLimit& limit)
+{
+    if (RefPtr renderer = rendererFor(identifier))
+        renderer->setPlatformDynamicRangeLimit(limit);
+}
+
+void RemoteAudioVideoRendererProxyManager::setResourceOwner(RemoteAudioVideoRendererIdentifier identifier, const WebCore::ProcessIdentity& resourceOwner)
+{
+    if (RefPtr renderer = rendererFor(identifier))
+        renderer->setResourceOwner(resourceOwner);
+}
+
+void RemoteAudioVideoRendererProxyManager::flushAndRemoveImage(RemoteAudioVideoRendererIdentifier identifier)
+{
+    if (RefPtr renderer = rendererFor(identifier))
+        renderer->flushAndRemoveImage();
+}
+
+void RemoteAudioVideoRendererProxyManager::currentVideoFrame(RemoteAudioVideoRendererIdentifier identifier, CompletionHandler<void(std::optional<RemoteVideoFrameProxy::Properties>)>&& completionHandler) const
+{
+    if (RefPtr renderer = rendererFor(identifier)) {
+        std::optional<RemoteVideoFrameProxy::Properties> result;
+        if (RefPtr videoFrame = renderer->currentVideoFrame())
+            result = m_videoFrameObjectHeap->add(videoFrame.releaseNonNull());
+        completionHandler(WTFMove(result));
+        return;
+    }
+    completionHandler(std::nullopt);
```

Comment:
This is a little more edge case, but I'd still refactor this to be an early-return in most cases:

```suggestion
    RefPtr renderer = rendererFor(identifier);
    if (!renderer) {
        completionHandler(std::nullopt);
        return;
    }

   
    if (RefPtr videoFrame = renderer->currentVideoFrame()) {
        completionHandler(m_videoFrameObjectHeap->add(videoFrame.releaseNonNull()));
        return;
    }

    completionHandler(std::nullopt);    
```